### PR TITLE
BkRepoFS

### PR DESCRIPTION
**Pull request by felixncheng**
基于Spring WebFlux实现文件系统。使用Kotlin Coroutines实现Reactive。使用Kotlin Dsl配置SpringBoot。

### CODE REVIEW

**Comment by tencentblueking-adm** [comment]
[CLA assistant check](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-repo?pullRequest=106) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-repo?pullRequest=106) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bk-cla.bktencent.com:443/check/TencentBlueKing/bk-repo?pullRequest=106) it.</sub>

File: src/backend/fs/server/boot-fs-server/src/main/kotlin/com/tencent/bkrepo/fs/server/request/NodePageRequest.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.web.reactive.function.server.ServerRequest
+import org.springframework.web.reactive.function.server.queryParamOrNull
+
+class NodePageRequest(request: ServerRequest) : NodeRequest(request) {
+    val listOption: NodeListOption
+    init {
+        val includeFolder = request.queryParamOrNull("includeFolder").toBoolean()
+        listOption = NodeListOption(
+            includeFolder = includeFolder,
+            pageSize = 10000
```
Comment (cnlkl) [inline src/backend/fs/server/boot-fs-server/src/main/kotlin/com/tencent/bkrepo/fs/server/request/NodePageRequest.kt:40]: page 和page size没有从参数取

File: src/backend/fs/server/boot-fs-server/src/main/kotlin/com/tencent/bkrepo/fs/server/service/PermissionService.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import kotlinx.coroutines.reactor.awaitSingle
+import org.springframework.web.reactive.function.server.ServerRequest
+
+/**
+ * 权限服务
+ * */
+class PermissionService(
+    private val rAuthClient: RAuthClient
+) {
+    suspend fun checkPermission(request: ServerRequest, action: PermissionAction): Boolean {
```
Comment (cnlkl) [inline src/backend/fs/server/boot-fs-server/src/main/kotlin/com/tencent/bkrepo/fs/server/service/PermissionService.kt:45]: 这边相比目前Generic服务用permissionManager校验权限少了些检查项

File: src/backend/fs/server/boot-fs-server/src/main/kotlin/com/tencent/bkrepo/fs/server/request/NodePageRequest.kt (hunk 1 above)
Comment (felixncheng) [reply src/backend/fs/server/boot-fs-server/src/main/kotlin/com/tencent/bkrepo/fs/server/request/NodePageRequest.kt]: 这里是打算把所有节点拉取，所以不需要从请求中读取。这里更像是一个限制，最大拉取数量。我已修改代码，让它表达的更加具体。

File: src/backend/fs/server/boot-fs-server/src/main/kotlin/com/tencent/bkrepo/fs/server/service/PermissionService.kt (hunk 2 above)
Comment (felixncheng) [reply src/backend/fs/server/boot-fs-server/src/main/kotlin/com/tencent/bkrepo/fs/server/service/PermissionService.kt:45]: 文件系统面临的场景与与原先的应用场景有差，所以权限上没有之前那么复杂。现在可以先简单点做起。

File: src/backend/fs/server/boot-fs-server/src/main/kotlin/com/tencent/bkrepo/fs/server/handler/FileOperationsHandler.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            val repo = RepositoryCache.getRepoDetail(projectId, repoName)
+            val node = rRepositoryClient.getNodeDetail(projectId, repoName, fullPath).awaitSingle().data!!
+            val length = node.size
+            val httpRange = request.headers().range().firstOrNull()?.let {
+                val startPosition = it.getRangeStart(length)
+                val endPosition = it.getRangeEnd(length)
+                Range(startPosition, endPosition, length)
+            }
+            val range = httpRange ?: Range.full(length)
+            val artifactInputStream = storageService.load(node.sha256!!, range, repo.storageCredentials)
```
[the remark this answers is not in the document]
Comment (felixncheng) [reply src/backend/fs/server/boot-fs-server/src/main/kotlin/com/tencent/bkrepo/fs/server/handler/FileOperationsHandler.kt]: 感谢指出，已处理。

File: src/backend/fs/server/boot-fs-server/src/main/kotlin/com/tencent/bkrepo/fs/server/handler/FileOperationsHandler.kt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+ *
+ * 处理文件操作请求
+ * */
+class FileOperationsHandler(
+    private val storageService: StorageService,
+    private val rRepositoryClient: RRepositoryClient
+) {
+
+    suspend fun download(request: ServerRequest): ServerResponse {
+        println("enter file operation")
```
[the remark this answers is not in the document]
Comment (felixncheng) [reply src/backend/fs/server/boot-fs-server/src/main/kotlin/com/tencent/bkrepo/fs/server/handler/FileOperationsHandler.kt]: 已删除

**Review by cnlkl — APPROVED** [review]

**Review by owenlxu — APPROVED** [review]

**Pull request merged**